app/services/markdown_processor/parser.rb

```diff
@@ -20,7 +20,8 @@ def finalize(link_attributes: {})
       renderer = Redcarpet::Render::HTMLRouge.new(options)
       markdown = Redcarpet::Markdown.new(renderer, Constants::Redcarpet::CONFIG)
       catch_xss_attempts(@content)
-      escaped_content = escape_liquid_tags_in_codeblock(@content)
+      code_tag_content = convert_code_tags_to_triple_backticks(@content)
```

Comment:
After looking through the application, we found that the following public methods on the `MarkdownProcessor::Parser` are being called

- evaluate_markdown: https://github.com/forem/forem/blob/a5e6f7942c6baf92cfb8e2cb591259d2bb2931d5/app/models/event.rb#L26
- evaluate_limited_markdown: https://github.com/forem/forem/blob/82b7f6519d736a5c825a674551613ac44876bb7c/app/models/organization.rb#L129
- evaluate_inline_limited_markdown: https://github.com/forem/forem/blob/a5e6f7942c6baf92cfb8e2cb591259d2bb2931d5/app/models/poll.rb#L34
- evaluate_listings_markdown: https://github.com/forem/forem/blob/a5e6f7942c6baf92cfb8e2cb591259d2bb2931d5/app/models/listing.rb#L72

I just included one example for each public method to show it is being used somewhere.

At the moment, this code tag fix is only applied when the application uses the `finalize` public method.

Is this the only place we want to fix to be applied? Perhaps we want it everywhere that the `code` tag is available?

Reply:
@atsmith813 Has done some work on the Markdown service objects lately, maybe he wants to weigh in here.

Reply:
For the sake of this PR and fixing the issue, I think it's okay for it to only be in `finalize`. Admittedly, I'm not sure _why_ the other cases (Events, Orgs, Listings, etc.) are parsed differently. That's been a long-standing thing we do - likely some legacy edge cases.

@benhalpern do you happen to recall why those objects are parsed differently?

Reply:
Ok.Thanks. @Miguel-Enrique13 and I will open a different issue going over any clean up / refactoring of the `MarkdownProcessor::Parser` public methods.

Reply:
Sound good, @mkcode. Thanks!

Reply:
Opened an [issue](https://github.com/forem/forem/issues/12874) for this.